File: 0_Trabalho_Pratico_Padaria/Bakehouse/Bakehouse.Application/ViewObjects/Product/ProductVO.cs
using FluentResults;
using System;

namespace Bakehouse.App.ViewObjects.Product
{
    public class ProductVO
    {
        public int Id { get; set; }
        public string BarCode { get; set; }
        public string Description { get; set; }
        public double Quantity { get; set; }
        public double ValueUnitary { get; set; }
        public double MinQuantity { get; set; }
        public int UnitOfMeasurementId { get; set; }
        public int CategoryId { get; set; }

        public DateTime CreatedAt { get; set; }

        public Result Valid()
        {
            try
            {
                if (BarCode.Trim().Length > 150 || BarCode.Trim().Length <= 0)
                    return Result.Fail("Código de barras deve ter no máximo 150 caracteres");
                if (Description.Trim().Length > 250 || Description.Trim().Length <= 0)
                    return Result.Fail("Descrção deve ter no maximo 250 caracteres");
                if (Quantity < 0)
                    return Result.Fail("Quantidade deve ser maior que -1");
                if (ValueUnitary <= 0)
                    return Result.Fail("Valor unitário deve ser maior que 0");
                if (MinQuantity <= 0)
                    return Result.Fail("Quantidade mínima deve ser maior que 0");
                if (UnitOfMeasurementId <= 0)
                    return Result.Fail("Unidade de medida deve ser informada");
                if (CategoryId <= 0)
                    return Result.Fail("Categoria deve ser informada");

                return Result.Ok();
            }
            catch (Exception)
            {
                return Result.Fail("Ocorreu um erro inesperado");
            }
        }
    }
}

File: 0_Trabalho_Pratico_Padaria/BakehouseJpaHB/src/main/java/com/bakehouse/viewobjects/product/ProductVO.java
package com.bakehouse.viewobjects.product;

import com.bakehouse.domain.Category;
import com.bakehouse.domain.UnitOfMeasurement;
import com.bakehouse.helpers.Result;
import com.bakehouse.helpers.Validations;

public class ProductVO {
    private int id;
    private String description;
    private String barCode;
    private double quantity;
    private double minQuantity;
    private double valueUnitary;
    private Category category;
    private UnitOfMeasurement unitOfMeasurement;

    public ProductVO() {
    }

    public ProductVO(int id, String description, String barCode, double quantity, double minQuantity, double valueUnitary, Category category, UnitOfMeasurement unitOfMeasurement) {
        this.id = id;
        this.description = description;
        this.barCode = barCode;
        this.quantity = quantity;
        this.minQuantity = minQuantity;
        this.valueUnitary = valueUnitary;
        this.category = category;
        this.unitOfMeasurement = unitOfMeasurement;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getBarCode() {
        return barCode;
    }

    public void setBarCode(String barCode) {
        this.barCode = barCode;
    }

    public double getQuantity() {
        return quantity;
    }

    public void setQuantity(double quantity) {
        this.quantity = quantity;
    }

    public double getMinQuantity() {
        return minQuantity;
    }

    public void setMinQuantity(double minQuantity) {
        this.minQuantity = minQuantity;
    }

    public double getValueUnitary() {
        return valueUnitary;
    }

    public void setValueUnitary(double valueUnitary) {
        this.valueUnitary = valueUnitary;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public UnitOfMeasurement getUnitOfMeasurement() {
        return unitOfMeasurement;
    }

    public void setUnitOfMeasurement(UnitOfMeasurement unitOfMeasurement) {
        this.unitOfMeasurement = unitOfMeasurement;
    }
    
    public Result valid() {
        try {
            if (id < -1 || id == 0)
                return new Result("Id inválido", false);
            if (Validations.stringIsNullOrEmpty(description) || description.length() > 30) 
                return new Result("Descrição deve ser informada e conter até 30 caracteres", false);
            if (Validations.stringIsNullOrEmpty(barCode) || barCode.length() > 15)
                return new Result("Código de barras deve ser informada e conter até 15 caracteres", false);
            if (quantity <= 0)
                return new Result("Quantidade deve ser maior que zero", false);
            if (minQuantity <= 0)
                return new Result("Quantidade mínima deve ser maior que zero", false);
            if (valueUnitary <= 0)
                return new Result("Valor unitário deve ser maior que zero", false);
            if (category == null)
                return new Result("Categoria deve ser informada", false);
            if (unitOfMeasurement == null)
                return new Result("Unidade de medida deve ser informada", false);
            
            return new Result("", true);
        } catch (Exception ex) {
            return new Result("Falha inesperada ao validar entradas", false);
        }
    }
}